File: samples/discovery/ios/ios/ComponentBoxUI/Sources/ComponentBoxUI/components/Row.swift
//
// Created by Matt Ramotar on 3/27/22.
// Copyright (c) 2022 Dropbox, Inc. All rights reserved.
//

import Foundation
import SwiftUI
import componentbox

@available(iOS 14.0, *)
struct ComponentBoxUIRow: View, Identifiable {
    let id: String
    let components: [componentbox.Component]?
    let modifier: componentbox.Modifier?
    let horizontalArrangement: componentbox.Arrangement?
    let verticalAlignment: componentbox.Alignment?
    let action: String?
    let isLazy: Bool?


    var body: some View {

        if (self.components != nil) {
            if (self.isLazy == true) {
                ScrollView {
                    ForEach(self.components!, id: \.self) { component in
                        component.inflate()
                    }
                }
            } else {
                HStack {
                    ForEach(self.components!, id: \.self) { component in
                        component.inflate()
                    }
                }.frame(maxWidth: modifier?.fillMaxWidth == true ? .infinity : .none, alignment: .leading)
            }
        }
    }
}

File: componentbox/src/commonMain/kotlin/com/dropbox/componentbox/component/Row.kt
package com.dropbox.componentbox.component

import com.dropbox.componentbox.foundation.Events
import com.dropbox.componentbox.foundation.Alignment
import com.dropbox.componentbox.foundation.Arrangement
import com.dropbox.componentbox.foundation.Modifier
import com.dropbox.componentbox.impl.ComponentBoxRow

interface Row : Component.Box {
    var horizontalArrangement: Arrangement.Horizontal?
    var verticalAlignment: Alignment.Vertical?

    companion object {
        fun of(
            modifier: Modifier? = null,
            events: Events? = null,
            components: MutableList<Component>? = null,
            horizontalArrangement: Arrangement.Horizontal? = null,
            verticalAlignment: Alignment.Vertical? = null
        ): Row = ComponentBoxRow(
            modifier = modifier,
            events = events,
            components = components,
            horizontalArrangement = horizontalArrangement,
            verticalAlignment = verticalAlignment
        )
    }
}
